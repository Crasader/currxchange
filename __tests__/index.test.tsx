import { shallow } from "enzyme";
import * as React from "react";
import renderer from "react-test-renderer";

import App from "../pages/index";

const props = {rates: { EUR: 2, GBP: 3, USD: 1}};
describe("With Enzyme", () => {
  it("App shows the header", () => {
    const app = shallow(<App {...props} />);

    expect(app.find("h1").text()).toEqual("Exchange");
  });

  it("App updates the input text when changing the value", () => {
    const app = shallow(<App {...props} />);

    expect(app.find("input").props().value).toEqual(1);
    app.find("input") .simulate("change", {target: { value: 10 }});
    expect(app.find("input").props().value).toEqual(10);
  });
});

describe("With Snapshot Testing", () => {
  it("App shows the header", () => {
    const component = renderer.create(<App {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
