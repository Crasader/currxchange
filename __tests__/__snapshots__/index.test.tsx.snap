// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows Hello world! 1`] = `
<div>
  <h1>
    Exchange
  </h1>
  <div>
    <select
      onChange={[Function]}
      value="USD"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value={1}
    />
  </div>
  <div>
    <select
      onChange={[Function]}
      value="EUR"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
  <div>
    <div>
      EUR
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing App shows the header 1`] = `
<div>
  <h1>
    Exchange
  </h1>
  <div>
    <select
      onChange={[Function]}
      value="USD"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value={1}
    />
  </div>
  <div>
    <select
      onChange={[Function]}
      value="EUR"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="GBP"
      >
        GBP
      </option>
      <option
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
  <div>
    <div>
      EUR
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;
